# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/Realm.h
# Commit: ec72f104edbf
# Full Hash: ec72f104edbfe88e22e2ebd778cc7890c21fd2ce
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2021-03-06 09:33:16
# Regressor Bug: 1689413
# File Overlap Count: 1
# Description:
#   Bug 1689413 part 9 - Move proto from ObjectGroup to BaseShape. r=jonco
#   
#   The ObjectGroup still stores the proto as well but we don't really use it anymore.
#   This is simpler than removing the group's proto now because that would result in
#   "empty" groups and the group table then doesn't make sense anymore.
# ==============================================================================

diff -r cad616508f73 -r ec72f104edbf js/src/vm/Realm.h
--- a/js/src/vm/Realm.h	Sat Mar 06 01:05:22 2021 +0000
+++ b/js/src/vm/Realm.h	Sat Mar 06 01:05:22 2021 +0000
@@ -102,7 +102,7 @@
     for (size_t i = 0; i < NumEntries; i++) {
       const Entry& entry = entries_[i];
       if (entry.group && entry.shape->getObjectClass() == clasp &&
-          entry.group->proto() == proto) {
+          entry.group->protoDeprecated() == proto) {
         *group = entry.group;
         *shape = entry.shape;
         return true;